aws/resource_aws_elastic_beanstalk_application.go

```diff
@@ -89,67 +88,75 @@ func resourceAwsElasticBeanstalkApplicationDescriptionUpdate(beanstalkConn *elas
 }
 
 func resourceAwsElasticBeanstalkApplicationRead(d *schema.ResourceData, meta interface{}) error {
-	a, err := getBeanstalkApplication(d, meta)
+	conn := meta.(*AWSClient).elasticbeanstalkconn
+
+	var app *elasticbeanstalk.ApplicationDescription
+	err := resource.Retry(30*time.Second, func() *resource.RetryError {
+		var err error
+		app, err = getBeanstalkApplication(d.Id(), conn)
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		if app == nil {
+			if d.IsNewResource() {
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
 	if err != nil {
 		return err
 	}
-	if a == nil {
-		return err
-	}
 
-	d.Set("name", a.ApplicationName)
-	d.Set("description", a.Description)
+	d.Set("name", app.ApplicationName)
+	d.Set("description", app.Description)
 	return nil
 }
 
 func resourceAwsElasticBeanstalkApplicationDelete(d *schema.ResourceData, meta interface{}) error {
 	beanstalkConn := meta.(*AWSClient).elasticbeanstalkconn
 
-	a, err := getBeanstalkApplication(d, meta)
-	if err != nil {
-		return err
-	}
-	_, err = beanstalkConn.DeleteApplication(&elasticbeanstalk.DeleteApplicationInput{
+	_, err := beanstalkConn.DeleteApplication(&elasticbeanstalk.DeleteApplicationInput{
 		ApplicationName: aws.String(d.Id()),
 	})
 	if err != nil {
 		return err
```

Comment:
Future note: we should add `if isAWSErr(err, "InvalidBeanstalkAppID.NotFound", "") {` here if necessary to immediately `return nil`

Reply:
Sure, I can add that 👌

Reply:
Actually this error will never be returned from that method, per [docs](https://docs.aws.amazon.com/elasticbeanstalk/latest/api/API_DeleteApplication.html) and my own experiments. You get `200 OK` if you try to delete an application which doesn't exist.